src/api_server/src/lib.rs

```diff
@@ -208,6 +213,7 @@ impl ApiServer {
         let response = ParsedRequest::convert_to_response(&vmm_outcome);
 
         if vmm_outcome.is_ok() {
+            self.vmm_shared_info.write().unwrap().state = vmm_new_state;
```

Comment:
I see `vmm_shared_info` is no longer shared between `vmm` and `api_server` anymore.

You could change it from:
```diff
pub struct ApiServer {
    /// MMDS info directly accessible from the API thread.
    mmds_info: Arc<Mutex<Mmds>>,
-   /// VMM instance info directly accessible from the API thread.
-   vmm_shared_info: Arc<RwLock<InstanceInfo>>,
+   /// Firecracker instance info exposed through API.
+   instance_info: InstanceInfo,
    /// Sender which allows passing messages to the VMM.
    api_request_sender: mpsc::Sender<ApiRequest>,
    /// Receiver which collects messages from the VMM.
    vmm_response_receiver: mpsc::Receiver<ApiResponse>,
    /// FD on which we notify the VMM that we have sent at least one
    /// `VmmRequest`.
    to_vmm_fd: EventFd,
}
```

And only change `state` here in api_server like you did for `Paused/Running`